junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java

```diff
@@ -1057,6 +1058,40 @@ public static boolean isMethodPresent(Class<?> clazz, Predicate<Method> predicat
 		// @formatter:on
 	}
 
+	public static List<Field> getAnnotatedFields(Class<?> clazz, Class<? extends Annotation> annotation) {
+		return getFields(clazz).stream().filter(f -> f.isAnnotationPresent(annotation)).collect(toList());
+	}
+
+	public static List<Object> getAnnotatedFieldValues(Object object, Class<? extends Annotation> annotation) {
+		return getAnnotatedFields(object.getClass(), annotation).stream().map(f -> {
+			try {
+				return f.get(object);
+			}
+			catch (IllegalAccessException e) {
+				throw ExceptionUtils.throwAsUncheckedException(e);
+			}
+		}).collect(toList());
```

Comment:
Rather than:

```java
}).collect(toList());
```

...how about the following to keep consistency with the other methods in this class?

```java
})
// unmodifiable since returned by public, non-internal method(s)
.collect(toUnmodifiableList());
```